roxctl/common/environment-impl.go

```diff
@@ -16,15 +19,42 @@ type cliEnvironmentImpl struct {
 	colorfulPrinter printer.ColorfulPrinter
 }
 
-// NewCLIEnvironment creates a new CLI environment with the given IO and common.RoxctlHTTPClient
-func NewCLIEnvironment(io IO, c printer.ColorfulPrinter) Environment {
+var (
+	singleton Environment
+	once      sync.Once
+)
+
+// NewTestCLIEnvironment creates a new CLI environment with the given IO and common.RoxctlHTTPClient.
+// It should be only used within tests.
+func NewTestCLIEnvironment(_ *testing.T, io IO, c printer.ColorfulPrinter) Environment {
 	return &cliEnvironmentImpl{
 		io:              io,
 		colorfulPrinter: c,
 		logger:          NewLogger(io, c),
 	}
 }
 
+// CLIEnvironment creates a new default CLI environment.
+func CLIEnvironment() Environment {
```

Comment:
I don't like this approach as it makes our code not testable :(
We should have a single instance of env and pass it everywhere where needed